components/badge.jsx

```diff
@@ -0,0 +1,66 @@
+var React       = require('react'),
+    classnames  = require('classnames'),
+    ImageTag    = require('./imagetag.jsx');
+
+var Badge = React.createClass({
+    propTypes: {
+        title: React.PropTypes.string.isRequired,
+        status: React.PropTypes.string,
+        icon: React.PropTypes.string.isRequired,
+        icon2x: React.PropTypes.string.isRequired
+    },
+    /**
+     * returning the Status HTML
+     * @param statusVal
+     * @returns ReactElement
+     */
+    status: function (statusVal) {
+
+        if (statusVal === undefined) {
+            return '';
+        }
+        else{
+            statusVal = statusVal.toLowerCase();
+        }
+
+        var statusIcons = {
+            pending : 'fa-clock-o',
+            achieved: 'fa-check',
+            eligible: 'fa-pencil'
+        };
+
+        var statusClasses = classnames('fa', 'fa-fw', statusIcons[statusVal]);
+        var statusIcon = <i className={statusClasses} />;
+
+        var classNames = 'label label-default ' + statusVal;
+
+        return (
+            <div className="status">
+                <div className={classNames}>
+                    <span>{ statusIcon }</span>{statusVal}
+                </div>
+            </div>
+        );
+
+    },
+    render: function (){
+
+        var statusElement = "";
+        if( this.props.status ){
+            statusElement = this.status( this.props.status );
```

Comment:
because of how `this.status` now returns the empty string if there is no `this.props.status` value top work with, we don't need the `if` anymore. We can use a one line `var statusElement = this.status(this.props.status)`, although since the `this.status` function is responsible for simply saying "what the status is", we can even say this:
```
var statusElement = this.status();
```
and then rather than taking an argument, have `this.status` responsible for checking:
```
status: function() {
  if (!this.props.status) {
    return '';
  }
  var statusVal = this.props.status.toLowerCase();
  ...
}
```